.cirrus.yml

```diff
@@ -0,0 +1,63 @@
+test_task:
+  matrix:
+    - container:
+        matrix:
+          - image: python:3.6
+          - image: python:3.7
+          - image: python:3.8-rc
+          - image: pypy:3.6
+    - windows_container:
+        matrix:
+          # Python 3.6 container was not built for windows
+          # - image: python:3.6
+          - image: python:3.7
+          - image: python:3.8-rc
+  setup_script:
+    - pip install -r requirements/tests.txt
+    - pip install .
+  test_script:
+    - xonsh run-tests.xsh
+
+
+mac_test_task:
+  osx_instance:
+    image: mojave-base
+  env:
+    PATH: "$HOME/miniconda/bin:$PATH"
+    matrix:
+      - PYTHON: "3.6"
+      - PYTHON: "3.7"
+      - PYTHON: "3.8"
+  conda_script:
+    - wget "https://repo.continuum.io/miniconda/Miniconda3-latest-MacOSX-x86_64.sh" -O miniconda.sh;
+    - bash miniconda.sh -b -p $HOME/miniconda;
+    - conda config --set always_yes yes --set changeps1 no
+    - conda update -q conda
+    - conda info -a
+  setup_script:
+    - pip install -r requirements/tests.txt
+    - pip install .
+  test_script:
+    - xonsh run-tests.xsh
+
+
+black_task:
+  container:
+    image: python:3
+  setup_script:
+    - pip install black==18.9b0
+  lint_script:
+    - black --check --exclude=xonsh/ply/ xonsh/ xontrib/
+
+
+docs_task:
+  container:
+    image: python:3
+  setup_script:
+    - pip install -r requirements/docs.txt
+  build_script:
+    - cd docs && make html
+  upload_artifacts:
+    path: docs/_build/*
+
+# TODO: Docs deploy
```

Comment:
I think doctr really only works with travis

Reply:
You're right. I'll have to look at it more closely.

Reply:
I suspect it's mostly:

0. Embed authn secret
1. Make shallow checkout of target repo
2. Copy docs build
3. Push build

I would be interested in doing some git manipulations so that doc publication tracks releases, though.